ibis/pyspark/compiler.py

```diff
@@ -211,12 +250,12 @@ def compile_aggregation(t, expr, scope, **kwargs):
 
     if op.by:
         context = AggregationContext.GROUP
-        aggs = [t.translate(m, scope, context=context) for m in op.metrics]
+        aggs = [_compile_agg(t, m, scope, context) for m in op.metrics]
         bys = [t.translate(b, scope) for b in op.by]
         return src_table.groupby(*bys).agg(*aggs)
     else:
         context = AggregationContext.ENTIRE
-        aggs = [t.translate(m, scope, context=context) for m in op.metrics]
+        aggs = [_compile_agg(t, m, scope, context) for m in op.metrics]
```

Comment:
maybe pull `aggs` above the if/else

Reply:
The `aggs` needs to be compiled with the right context (e.g. `GROUP` or `ENTIRE`).

Reply:
oh, right. duh